pkg/operator/ceph/cluster/rbd/controller.go

```diff
@@ -278,5 +278,6 @@ func (r *ReconcileCephRBDMirror) updateStatus(client client.Client, name types.N
 		logger.Errorf("failed to set rbd mirror %q status to %q. %v", rbdMirror.Name, status, err)
 		return
 	}
+	rbdMirror.Status.ObservedGeneration = rbdMirror.ObjectMeta.Generation
```

Comment:
The generation needs to be applied from a different object. Here it was retrieved from line 262 in this method. Instead, we want to use the generation from the object retrieved at the beginning of the reconcile on [line 154](https://github.com/rook/rook/blob/master/pkg/operator/ceph/cluster/rbd/controller.go#L154). That will represent the correct generation that was reconciled. We want to avoid setting the generation retrieved on line 262 because the CR might have a different generation than when the reconcile started if the user quickly edited the CR again while a reconcile was in progress.

Reply:
I agree with @travisn .

In another PR, I was even saying that maybe we shouldn't re-fetch object while updating status as we could possibly be updating status on an object generation that operator hasn't reconciled yet. Or am I wrong about this?

cc @leseb